proposals/text/4084-agent-demo.md

```diff
@@ -0,0 +1,191 @@
+- Title: Agent tools
+- Decision driver: Vladimir Blagojevic (with Julian Risch)
+- Start Date: 2023-02-06
+- Proposal PR: https://github.com/deepset-ai/haystack/pull/4084
+
+
+# Summary
```

Comment:
Overall I would keep the summary minimal and not introduce the concept of `Agent` in this proposal. I would make reference to the original Agent proposal and describe only relevant things to the Agent demo.